core/src/test/java/org/apache/calcite/sql/test/SqlOperatorBaseTest.java

```diff
@@ -5577,6 +5577,46 @@ private void checkNullOperand(SqlTester tester, String op) {
     }
   }
 
+  @Test public void testLastDayFunc() {
+    tester.setFor(SqlStdOperatorTable.LAST_DAY);
+    tester.checkScalar("last_day(DATE '2019-02-10')",
```

Comment:
Can tests more edge cases like below (last day of the month, leap year etc.) : 
1. `DATE '2019-02-28'`
2.  `DATE '2020-02-20'` /  `DATE '2020-02-29'` (leap years: 2012 / 2020 / 2076). 
2. `DATE '2019-12-31'`
3. `DATE '2019-01-01'`
4. `DATE '2019-06-30'`
4. `DATE '9999-12-31'`

Reply:
What about dates before 1970-01-01 ?

Reply:
I will add more tests about dates before 1970-01-01 .